realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1070,17 +1069,59 @@ public void executeTransaction(Transaction transaction) {
      * Similar to {@link #executeTransaction(Transaction)} but runs asynchronously from a worker thread.
      *
      * @param transaction {@link io.realm.Realm.Transaction} to execute.
-     * @param callback optional, to receive the result of this query.
      * @return a {@link RealmAsyncTask} representing a cancellable task.
      * @throws IllegalArgumentException if the {@code transaction} is {@code null}, or if the realm is opened from another thread.
      */
-    public RealmAsyncTask executeTransaction(final Transaction transaction, final Transaction.Callback callback) {
```

Comment:
To follow our normal deprecation policies we would need to deprecate this for 2 releases instead of removing it outright.